src/Simulation/TargetDefinitions/Decompositions/MeasureWithNoReuse.qs

```diff
@@ -39,27 +40,120 @@ namespace Microsoft.Quantum.Intrinsic {
+            return MResetZ(q);
+        }
+        elif Length(bases) == 4 {
+            use q = Qubit();
+            within {
+                H(q);
+            }
+            apply {
+                JointMeasureHelper(bases[0], q, qubits[0]);
+                JointMeasureHelper(bases[1], q, qubits[1]);
+                JointMeasureHelper(bases[2], q, qubits[2]);
+                JointMeasureHelper(bases[3], q, qubits[3]);
+            }
+            return MResetZ(q);
+        }
+        elif Length(bases) == 5 {
+            use q = Qubit();
+            within {
+                H(q);
             }
+            apply {
+                JointMeasureHelper(bases[0], q, qubits[0]);
+                JointMeasureHelper(bases[1], q, qubits[1]);
+                JointMeasureHelper(bases[2], q, qubits[2]);
+                JointMeasureHelper(bases[3], q, qubits[3]);
+                JointMeasureHelper(bases[4], q, qubits[4]);
+            }
+            return MResetZ(q);
+        }
+        elif Length(bases) == 6 {
+            use q = Qubit();
+            within {
+                H(q);
+            }
+            apply {
+                JointMeasureHelper(bases[0], q, qubits[0]);
+                JointMeasureHelper(bases[1], q, qubits[1]);
+                JointMeasureHelper(bases[2], q, qubits[2]);
+                JointMeasureHelper(bases[3], q, qubits[3]);
+                JointMeasureHelper(bases[4], q, qubits[4]);
+                JointMeasureHelper(bases[5], q, qubits[5]);
+            }
+            return MResetZ(q);
+        }
+        elif Length(bases) == 7 {
+            use q = Qubit();
+            within {
+                H(q);
+            }
+            apply {
+                JointMeasureHelper(bases[0], q, qubits[0]);
+                JointMeasureHelper(bases[1], q, qubits[1]);
+                JointMeasureHelper(bases[2], q, qubits[2]);
+                JointMeasureHelper(bases[3], q, qubits[3]);
+                JointMeasureHelper(bases[4], q, qubits[4]);
+                JointMeasureHelper(bases[5], q, qubits[5]);
+                JointMeasureHelper(bases[6], q, qubits[6]);
+            }
+            return MResetZ(q);
+        }
+        elif Length(bases) == 8 {
+            use q = Qubit();
+            within {
+                H(q);
+            }
+            apply {
+                JointMeasureHelper(bases[0], q, qubits[0]);
+                JointMeasureHelper(bases[1], q, qubits[1]);
+                JointMeasureHelper(bases[2], q, qubits[2]);
+                JointMeasureHelper(bases[3], q, qubits[3]);
+                JointMeasureHelper(bases[4], q, qubits[4]);
+                JointMeasureHelper(bases[5], q, qubits[5]);
+                JointMeasureHelper(bases[6], q, qubits[6]);
+                JointMeasureHelper(bases[7], q, qubits[7]);
+            }
+            return MResetZ(q);
+        }
+        else {
+            fail "Too many qubits specified in call to Measure.";
         }
     }
```

Comment:
(I can be wrong but) I have an impression that the changes in [Measure.qs](https://github.com/microsoft/qsharp-runtime/pull/997/files#diff-f4fcfc529ef943522c75549272dbcec2437ad6803109db9a0ec6c917d3df72b5) and [MeasureWithNoReuse.qs](https://github.com/microsoft/qsharp-runtime/pull/997/files#diff-716e9e692507d2246f5d7a51db7ac8b6c13549bcd9f0ac0da16abf7d9364da12) have a significant common part that can be extracted into a separate file and placed into an operation invoked from both files.

Reply:
Hmm... you're right. As I was working on these initially they had more differences, but now the only difference is in the `Length(bases) == 1` case. I'll pull the rest out to a common utility operation.